x-pack/plugins/endpoint/public/embeddables/resolver/store/data/selectors.ts

```diff
@@ -0,0 +1,302 @@
+  return eventType(event) === 'processCreated' || eventType(event) === 'processRan';
+}
+
+export function eventType(event) {
+  const {
+    data_buffer: { event_type_full: type, event_subtype_full: subType },
+  } = event;
+
+  if (type === 'process_event') {
+    if (subType === 'creation_event' || subType === 'fork_event' || subType === 'exec_event') {
+      return 'processCreated';
+    } else if (subType === 'already_running') {
+      return 'processRan';
+    } else if (subType === 'termination_event') {
+      return 'processTerminated';
+    } else {
+      return 'unknownProcessEvent';
+    }
+  } else if (type === 'alert_event') {
+    return 'processCausedAlert';
+  }
+  return 'unknownEvent';
+}
+
+function uniquePidForProcess(event) {
+  return event.data_buffer.node_id;
+}
+
+function uniqueParentPidForProcess(event) {
+  return event.data_buffer.source_id;
+}
+
+function yHalfWayBetweenSourceAndTarget(sourcePosition: Vector2, targetPosition: Vector2) {
+  return sourcePosition[1] + (targetPosition[1] - sourcePosition[1]) / 2;
+}
+
+function childrenOfProcessFromGraphableProcessesPidMaps(graphableProcessesPidMaps, parentProcess) {
+  const uniqueParentPid = uniquePidForProcess(parentProcess);
+  const children = graphableProcessesPidMaps.processesByUniqueParentPid.get(uniqueParentPid);
+  return children === undefined ? [] : children;
+}
+
+function parentOfProcessFromGraphableProcessesPidMaps(graphableProcessesPidMaps, childProcess) {
+  const uniqueParentPid = uniqueParentPidForProcess(childProcess);
+  return graphableProcessesPidMaps.processesByUniquePid.get(uniqueParentPid);
+}
+
+function isProcessOnlyChildFromGraphableProcessPidMaps(graphableProcessesPidMaps, childProcess) {
+  const parentProcess = parentOfProcessFromGraphableProcessesPidMaps(
+    graphableProcessesPidMaps,
+    childProcess
+  );
+  return (
+    childrenOfProcessFromGraphableProcessesPidMaps(graphableProcessesPidMaps, parentProcess)
+      .length === 1
+  );
+}
+
+export function graphableProcesses(state: ResolverState) {
+  return dataSelector(state).filter(isGraphableProcess);
+}
+
+const graphableProcessesPidMaps = createSelector(
+  graphableProcesses,
+  function graphableProcessesPidMaps(
+    /* eslint-disable no-shadow */
+    graphableProcesses
+    /* eslint-disable no-shadow */
+  ) {
+    const processesByUniqueParentPid = new Map();
+    const processesByUniquePid = new Map();
+
+    for (const process of graphableProcesses) {
+      processesByUniquePid.set(uniquePidForProcess(process), process);
+      const uniqueParentPid = uniqueParentPidForProcess(process);
+      if (processesByUniqueParentPid.has(uniqueParentPid)) {
+        processesByUniqueParentPid.get(uniqueParentPid).push(process);
+      } else {
+        processesByUniqueParentPid.set(uniqueParentPid, [process]);
+      }
+    }
+
+    return {
+      processesByUniqueParentPid,
+      processesByUniquePid,
+    };
+  }
+);
+const widthOfProcessSubtrees = createSelector(
```

Comment:
:question: In comments, some explanation of _why_ you are doing this would be helpful to the reader.